openhands/cli/utils.py

```diff
@@ -236,13 +236,47 @@ def is_first_time_user() -> bool:
     return not openhands_dir.exists()
 
 
-def get_bash_profile_path() -> Path:
-    """Get the path to the user's bash profile file.
+def get_shell_config_path() -> Path:
+    """Get the path to the user's shell configuration file based on their default shell.
+
+    Uses shellingham to detect the current shell and returns the appropriate config file path.
 
     Returns:
-        Path to the bash profile file (prefers .bashrc, falls back to .bash_profile).
+        Path to the shell configuration file.
     """
     home = Path.home()
+
+    # Default config paths for different shells
+    shell_configs = {
+        'bash': ['.bashrc', '.bash_profile'],
+        'zsh': ['.zshrc'],
+        'fish': ['.config/fish/config.fish'],
+        'csh': ['.cshrc'],
+        'tcsh': ['.tcshrc'],
+        'ksh': ['.kshrc'],
+        'powershell': ['.config/powershell/Microsoft.PowerShell_profile.ps1'],
+    }
+
+    # Try to detect the shell using shellingham
+    try:
+        import shellingham
+
+        shell_name, _ = shellingham.detect_shell()
+
+        # Get config files for the detected shell
+        if shell_name in shell_configs:
+            for config_file in shell_configs[shell_name]:
+                config_path = home / config_file
+                if config_path.exists():
+                    return config_path
+
+            # If no existing config file found, return the first one in the list
+            return home / shell_configs[shell_name][0]
```

Comment:
@OpenHands We should throw an error here saying your shell is not supported, and tell them the list of shells that's supported. When you are done fixing this issue, pls run `poetry lock` to update the lock file